### PR TITLE
Bug 1283457 - Update username event should spread to the stores

### PR DESCRIPTION
**Pull request by piatra**
Implemented using LoopAPI as described in [comment](https://bugzilla.mozilla.org/show_bug.cgi?id=1279042#c3) propagates the event from the panel to the text chat store.

### CODE REVIEW

**Comment by piatra** [comment]
@Standard8 @fcampo please review.
Locally functional tests are passing I suspect some task cluster shenanigans but let me know if issue reproduces.

**Comment by Standard8** [comment]
> Locally functional tests are passing I suspect some task cluster shenanigans but let me know if issue reproduces.

That's an intermittent, I've just filed it as [bug 1284805](https://bugzilla.mozilla.org/show_bug.cgi?id=1284805)

File: add-on/panels/js/roomStore.js [hunk 1]
```diff
@@ -124,6 +125,13 @@ loop.store = loop.store || {};
     },
 
     /**
+     * Pass a sharedAction.SetOwnDisplayName from chrome to content space.
+     */
+    setOwnDisplayName: function(actionData) {
+      loop.request("SetOwnDisplayName", actionData);
```
Comment (fcampo) [inline add-on/panels/js/roomStore.js]: as the current flow is `panel.jsx` requesting a change of preferences, and then dispatching the action that `roomStore.js` is receiving, just to request another API call, maybe we can change it for panel requesting this new `SetOwnDisplayName` instead, and the API setting the preference right before broadcasting the message, so no changes on roomStore are needed.
Although I'm not strong on this, it seems, at first glance, less messy for the action/messages back and forth.

File: shared/js/textChatStore.js [hunk 2]
```diff
@@ -49,6 +49,9 @@ loop.store.TextChatStore = (function() {
       }
 
       this._dataDriver = options.dataDriver;
+
+      loop.subscribe("Panel:SetOwnDisplayName",
+                     this.updateOwnDisplayName.bind(this));
```
Comment (fcampo) [inline shared/js/textChatStore.js:54]: the `participantStore` should also subscribe to this event, so the presence is updated on the sidebar when the name changes.

File: add-on/panels/js/roomStore.js (hunk 1 above)
Comment (piatra) [inline add-on/panels/js/roomStore.js]: That makes sense. This patch has no changes to `panel.jsx` the action dispatch was already happening there so I tried to follow that pattern.

**Comment by piatra** [comment]
@fcampo updated.

File: shared/js/participantStore.js [hunk 3]
```diff
@@ -72,6 +75,15 @@ loop.store.ParticipantStore = function() {
     },
 
     /**
+     * Receives action from chrome space.
+     *
+     * @param {sharedActions.SetOwnDisplayName[]} actionData.data
+     */
+    updateOwnDisplayName(actionData) {
+      this.setOwnDisplayName(actionData.data[0]);
```
Comment (fcampo) [inline shared/js/participantStore.js:83]: adding a call to update the room info would trigger a change on the presence indicator for all the guests connected to the room

`this.updateRoomInfo({ userId: this._currentUserId });`

**Pull request merged**